Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/BoundingBox/BoundingBox.cs

```diff
@@ -1536,158 +1566,121 @@ private void SetBoundingBoxCollider()
+                    if (AddRendererBoundsToTarget(rendererBoundsByTransform)) { continue; }
+                    if (boundsCalculationMethod == BoundsCalculationMethod.RendererOnly) { continue; }
                 }
 
-                boundsMethod = BoundsCalculationMethod.Renderers;
-                return bounds;
+                // Do the collider for the one case that we chose RendererOverCollider and did not find a renderer
+                AddColliderBoundsToTarget(colliderByTransform);
             }
 
-            MeshFilter[] meshFilters = Target.GetComponentsInChildren<MeshFilter>();
-            if (meshFilters.Length > 0)
+            if (colliderCorners.Count == 0) { return new Bounds(); }
+
+            Transform targetTransform = Target.transform;
+
+            Bounds finalBounds = new Bounds(targetTransform.InverseTransformPoint(colliderCorners[0]), Vector3.zero);
+
+            for (int i = 1; i < colliderCorners.Count; i++)
             {
-                //Mesh.bounds is local space bounding volume
-                bounds.size = meshFilters[0].mesh.bounds.size;
-                bounds.center = meshFilters[0].mesh.bounds.center;
-                for (int i = 0; i < meshFilters.Length; ++i)
-                {
-                    bounds.Encapsulate(meshFilters[i].mesh.bounds);
-                }
-                if (bounds.size != Vector3.zero)
-                {
-                    bounds.center = Target.transform.position;
-                    boundsMethod = BoundsCalculationMethod.MeshFilters;
-                    return bounds;
-                }
+                finalBounds.Encapsulate(targetTransform.InverseTransformPoint(colliderCorners[i]));
             }
 
-            BoxCollider boxCollider = Target.AddComponent<BoxCollider>();
-            bounds = boxCollider.bounds;
-            Destroy(boxCollider);
-            boundsMethod = BoundsCalculationMethod.Collider;
-            return bounds;
+            return finalBounds;
         }
 
-        private Bounds GetSingleObjectBounds(GameObject gameObject)
+        private bool AddRendererBoundsToTarget(KeyValuePair<Transform, Bounds> rendererBoundsByTarget)
         {
-            Bounds bounds = new Bounds(Vector3.zero, Vector3.zero);
-            BoxCollider boxCollider;
-            boxCollider = gameObject.GetComponent<BoxCollider>();
-            if (boxCollider == null)
-            {
-                boxCollider = gameObject.AddComponent<BoxCollider>();
-                bounds = boxCollider.bounds;
-                DestroyImmediate(boxCollider);
-            }
-            else
-            {
-                bounds = boxCollider.bounds;
-            }
+            if (rendererBoundsByTarget.Key == null) { return false; }
+
+            Vector3[] cornersToWorld = null;
+            rendererBoundsByTarget.Value.GetCornerPositions(rendererBoundsByTarget.Key, ref cornersToWorld);
+            colliderCorners.AddRange(cornersToWorld);
```

Comment:
To help with understandability in future, could you please add comment to colliderCorners explaining what that variable represents?

Reply:
I rephrased it, but the variable is explained where it's declared.